### PR TITLE
sc-399/poetry st_lib

### PR DESCRIPTION
**Pull request by epcAnker**
Special things i had to do:

buildui - the ui files are compiled in setup.py.  I created a file "buildui.py" to do the same thing.  I added this to both ST & STlib (jury is still out on if i want to call buildui in stlib from st or if i want it to do the work itself?  See code.. w/ comments).  This needed to be added to the readme for manual instructions (I added it to the ci).

having issues pyqt5 tools in poetry

Removed all romp items, requirements, boots, setup.py

all venv -> .venv (poetry uses .venv)

------

To use:

first install poetry  
for powershell: `(Invoke-WebRequest -Uri https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py -UseBasicParsing).Content | python -`

linux: `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -`

`poetry install` should install dependencies and scripts

`poetry run epyqlib` (or activate the .venv)  
 

---

(will put a poetry guide on confluence)

If you want the .venv to install in your directory:

`poetry config virtualenvs.in-project = true`  
`poetry config virtualenvs.path = "."`

`poetry config --list` shows all config.  mine looks like:

cache-dir = "C:\\\\Users\\\\Alex\\\\AppData\\\\Local\\\\pypoetry\\\\Cache"  
experimental.new-installer = true  
installer.parallel = true  
virtualenvs.create = true  
virtualenvs.in-project = true  
virtualenvs.path = "."

### CODE REVIEW

File: .circleci/prepare.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 $PYTHON boots.py create --group package
```
Comment (epcAnker) [inline .circleci/prepare.sh:1]: this may need to change from boots.

Comment (gordon-epc) [reply .circleci/prepare.sh:1]: I think this file and nearby files are the leftovers of a previous CI process and can be removed.

File: deploy_win.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 import sys
 
+sys.path.append(".\..\..\src")
```
Comment (epcAnker) [inline deploy_win.py:5]: This is hacky but it works

File: epyqlib/builduiepyqlib.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import alqtendpy.compileui
+import pathlib
+
```
Comment (epcAnker) [inline epyqlib/builduiepyqlib.py:3]: note: this file is ran from gridtied as well

File: deploy_win.py (hunk 2 above)
Comment (gordon-epc) [inline deploy_win.py:5]: This line seems strange. Can it be remediated? Is it still necessary? If so, can you add a comment explaining exactly what it is for?

File: pyproject.toml [hunk 4]
```diff
@@ -1,3 +1,96 @@
...
+poetry-dynamic-versioning = "^0.13.1"
+black = "20.8b1"
+pytest = "5.3.5"
+pytest-qt = "3.3.0"
+pytest-rerunfailures = "5.0"
+pytest-twisted = "<=1.13.4"
+pytest-xvfb = "1.2.0"
+
+[tool.poetry.scripts]
+collectdevices = 'epyqlib.collectdevices:main'
```
Comment (gordon-epc) [inline pyproject.toml:66]: Is there a reason the `outparsecheck` script from `setup.py` was not put on this list?
`outparsecheck = epyqlib.outparsecheck:main [dss]`

File: pyproject.toml [hunk 5]
```diff
@@ -1,3 +1,96 @@
...
+GitPython = "2.1.15"
+graham = "0.1.11"
+marshmallow = "2.16.3"
+natsort = "5.5.0"
+paho-mqtt = "1.4.0"
+Pint = "0.11"
+pyelftools = "0.26"
+PyQt5 = ">=5.13.0"
+##pyqt5-tools fails
+##pyqt5-tools = { version = "5.14.1.3", markers = "sys_platform == 'win32'" }
```
Comment (gordon-epc) [inline pyproject.toml:42]: I think we will need to look into why `pyqt5-tools` fails. It is used to design the UI, which is necessary for development. Perhaps it should go under the `tool.poetry.dev-dependencies` section? Also, I see `PyQt5Designer` below. Does that take the place of `pyqt5-tools`? Perhaps `PyQt5Designer` should go under `tool.poetry.dev-dependencies`?

File: pyproject.toml [hunk 6]
```diff
@@ -1,3 +1,96 @@
+[tool.poetry]
+name = "epyqlib"
+version = "0.0.0"
+description = ""
+authors = ["Alex Anker <alex.anker@epcpower.com>"]
+
```
Comment (gordon-epc) [inline pyproject.toml:6]: Add the license line.
```suggestion
license = "GPL-2.0-or-later"

```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: deploy_win.py (hunk 2 above)
Comment (epcAnker) [reply deploy_win.py:5]: STLib, the submodule of ST, tries importing ST.That would a "poetry add ../../" which isnt what we want

this is a hack to include it anyway.

(ill add this as a comment)

File: pyproject.toml (hunk 4 above)
Comment (epcAnker) [reply pyproject.toml:66]: ill add it but i dont think we actually use it.

File: pyproject.toml [hunk 7]
```diff
@@ -1,3 +1,103 @@
+[tool.poetry]
+name = "epyqlib"
+version = "2022.2.1.post35.dev0+6ede3f7"
```
Comment (gordon-epc) [inline pyproject.toml:3]:
```suggestion
version = "0.0.0"
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

**Review by bholewin-epc — APPROVED** [review]
Approved from my point of view. Please hold off on merging until Gordon gives his ok.

File: pyproject.toml (hunk 5 above)
Comment (gordon-epc) [reply pyproject.toml]: Addressed this by downgrading `pyqt5-tools`. The only other known option would have been to use a beta version of poetry. 6783fd6ef32a903b853a4cb286b3d9f2583c0972

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+Pint = "0.11"
+pyelftools = "0.26"
+PyQt5 = ">=5.13.0"
+pyqt5-tools = { version = "5.13.2.1.6rc1", markers = "sys_platform == 'win32'" }
+qt5reactor = "0.5"
+python-dateutil = "2.7.5"
+python-docx = "0.8.7"
+python-dotenv = "0.9.1"
+QtAwesome = "0.6.0"
+siphash-cffi = "^0.1.4"
```
Comment (gordon-epc) [inline pyproject.toml:49]: Should be locked at v0.1.4. Also need to switch from fstab to AWS CodeArtifact for hosting.
```suggestion
siphash-cffi = "0.1.4"
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 8 above)
Comment (epcAnker) [reply pyproject.toml:49]: Adding now.  not committing suggestion bc lock file needs to be update as well

Comment (gordon-epc) [reply pyproject.toml]: This is still specified to look for versions `^0.1.4`. Should be pinned at `0.1.4`.

**Review by gordon-epc — APPROVED** [review]

**Pull request merged**